assets/javascript/script.js

```diff
@@ -222,4 +223,49 @@ async function hightlightMarker(event) {
 }
 
 resultTable.on("mouseover", hightlightMarker)
-resultTable.on("mouseout", hightlightMarker)
\ No newline at end of file
+resultTable.on("mouseout", hightlightMarker)
+
+// search address functionality
+var apiKey = 'AIzaSyA5Zx1uReveYAhTFw1btOcdMgIMCY7GVNE';
+
+// function for autocomplete, Google Places API
+function initAutocomplete() {
+    var input = document.getElementById('addressInput');
+    var options = {
+        types: ['geocode']
+    };
+
+    var autocomplete = new google.maps.places.Autocomplete(input, options);
+
+    // When a place is selected, show it on the map
+    autocomplete.addListener('place_changed', function () {
+        var place = autocomplete.getPlace();
+        if (place.geometry) {
+            showOnMap(place.geometry.location);
+        }
+    });
+}
+
+// Load Google Maps Places API asynchronously
+function loadScript() {
```

Comment:
No need for this function at all, you can still call any libraries that you need by simple do google.maps.(that library).